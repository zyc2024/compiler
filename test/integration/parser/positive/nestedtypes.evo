

//pretend this is a really shitty programmer ig

type ColorPoint = { 
    int v;
}

type AuxPointData = {
    char[] name; 
    ColorPoint[] color;
}


type Dimension = {
    int numDims;
}

type NdPoint = {
    int[] coordinates;
    AuxPointData aux;
}