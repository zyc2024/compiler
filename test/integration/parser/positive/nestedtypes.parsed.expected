(
    ()
    (
        (ColorPoint ((v int))  )
        (AuxPointData ((name ([] char)) (color ([] ColorPoint)))  )
        (Dimension ((numDims int))  ) 
        (NdPoint ((coordinates ([] int) ) (aux AuxPointData)) )
    )
)